velox/exec/tests/RowNumberTest.cpp

```diff
@@ -310,4 +326,168 @@ TEST_F(RowNumberTest, memoryUsage) {
+      "facebook::velox::exec::Task::requestPauseLocked",
+      std::function<void(Task*)>(([&](Task* /*unused*/) {
+        taskPauseWaitFlag = false;
+        taskPauseWait.notifyAll();
+      })));
+
+  std::thread rowNumberThread([&]() {
+    auto task =
+        AssertQueryBuilder(duckDbQueryRunner_)
+            .spillDirectory(spillDirectory->path)
+            .config(core::QueryConfig::kSpillEnabled, true)
+            .config(core::QueryConfig::kRowNumberSpillEnabled, true)
+            .queryCtx(rowNumberQueryCtx)
+            .plan(PlanBuilder().values(vectors).rowNumber({"c0"}).planNode())
+            .assertResults(
+                "SELECT *, row_number() over (partition by c0) FROM tmp");
+    auto stats = task->taskStats().pipelineStats;
+    ASSERT_GT(stats[0].operatorStats[1].spilledBytes, 0);
+    ASSERT_EQ(stats[0].operatorStats[1].spilledPartitions, 4 * 2);
+  });
+
+  auto fakePool = fakeQueryCtx->pool()->addLeafChild(
+      "fakePool", true, FakeMemoryReclaimer::create());
+
+  arbitrationWait.await([&] { return !arbitrationWaitFlag.load(); });
+  ASSERT_TRUE(memoryManager->testingGrowPool(
+      fakePool.get(),
+      memoryManager->arbitrator()->capacity() -
+          rowNumberQueryCtx->pool()->currentBytes() + 1));
+
+  rowNumberThread.join();
+
+  waitForAllTasksToBeDeleted();
+}
+
+DEBUG_ONLY_TEST_F(RowNumberTest, recursiveSpill) {
+  std::vector<RowVectorPtr> vectors = createVectors(8, rowType_, fuzzerOpts_);
+  createDuckDbTable(vectors);
+  std::unique_ptr<memory::MemoryManager> memoryManager = createMemoryManager();
+
+  const auto spillDirectory = exec::test::TempDirectoryPath::create();
+
+  std::shared_ptr<core::QueryCtx> rowNumberQueryCtx;
+  rowNumberQueryCtx =
+      newQueryCtx(memoryManager, executor_, kMemoryCapacity * 2);
+  std::shared_ptr<core::QueryCtx> fakeQueryCtx =
+      newQueryCtx(memoryManager, executor_, kMemoryCapacity * 2);
+
+  folly::EventCount arbitrationWait;
+  std::atomic_bool arbitrationWaitFlag{true};
+  folly::EventCount taskPauseWait;
+  std::atomic_bool taskPauseWaitFlag{true};
+
+  std::atomic_int numInputs{0};
+  SCOPED_TESTVALUE_SET(
+      "facebook::velox::exec::Driver::runInternal::addInput",
+      std::function<void(Operator*)>(([&](Operator* op) {
+        if (op->operatorType() != "RowNumber") {
+          return;
+        }
+
+        if (++numInputs != 5) {
+          return;
+        }
+        arbitrationWaitFlag = false;
+        arbitrationWait.notifyAll();
+
+        // Wait for task pause to be triggered.
+        taskPauseWait.await([&] { return !taskPauseWaitFlag.load(); });
+      })));
+
+  SCOPED_TESTVALUE_SET(
+      "facebook::velox::exec::Task::requestPauseLocked",
+      std::function<void(Task*)>(([&](Task* /*unused*/) {
+        taskPauseWaitFlag = false;
+        taskPauseWait.notifyAll();
+      })));
+
+  std::atomic_int numOutputs{1};
+  SCOPED_TESTVALUE_SET(
+      "facebook::velox::exec::Driver::runInternal::getOutput",
```

Comment:
I think we just need a testValue of getOutput for this? And do we want a test to exceed the spill limit? Thanks!